### PR TITLE
fix: more json schema cases

### PR DESCRIPTION
**Pull request by kaposke**
## Description
Your daily dose of json schema mapping fixes. This time I exhausted my pool of real cases and they all look good.

I might do a second pass at the legacy converter, but hopefully we're done with the NangoModel one.

---

This PR refines the JSON Schema generation logic for Nango models to correctly handle nuanced TypeScript scenarios, including special types such as `any`, `any[]`, and `Record<string, any>`, as well as dynamic fields and empty models. It introduces a unified type mapping (`tsTypeMap`), ensures schema output avoids meaningless empty keys, and extends test coverage to new edge cases for improved correctness.

**Key Changes:**
• Added a unified `tsTypeMap` for handling TypeScript type-to-JSON-Schema translation, covering types like 'any', 'any[]', and 'Record<string, any>'.
• Modified schema generation to omit `properties` and `required` fields when empty, enhancing compliance with JSON Schema best practices.
• Improved support for dynamic fields (such as those with `__string` and `dynamic: true`), ensuring proper use of `additionalProperties` in generated schemas.
• Refined handling of models with no fields to avoid empty or extraneous keys.
• Expanded and updated unit tests and snapshots to cover new type mappings and edge behaviors.

**Affected Areas:**
• packages/utils/lib/jsonSchema/nangoModelToJsonSchema.ts
• packages/utils/lib/jsonSchema/nangoModelToJsonSchema.unit.test.ts
• packages/utils/lib/jsonSchema/__snapshots__/nangoModelToJsonSchema.unit.test.ts.snap

*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/utils/lib/jsonSchema/nangoModelToJsonSchema.ts [hunk 1]
```diff
@@ -138,11 +145,13 @@ function nangoFieldToJsonSchema(field: NangoModelField): JSONSchema7 {
...
+const tsTypeMap: Record<string, JSONSchema7> = {
     number: { type: 'number' },
     bigint: { type: 'number' },
     boolean: { type: 'boolean' },
     string: { type: 'string' },
     date: { type: 'string', format: 'date-time' },
-    undefined: { type: 'null' } // There's no json schema mapping for this.
+    undefined: { type: 'null' }, // There's no json schema mapping for this.
+    any: {},
+    'Record<string, any>': { type: 'object' }
```
Comment (TBonnin) [inline packages/utils/lib/jsonSchema/nangoModelToJsonSchema.ts:156]: would something like `Record<any, any>` or anything where the key is not a string ever representable?

**Review by TBonnin — APPROVED** [review]

**Review by bodinsamuel — APPROVED** [review]

File: packages/utils/lib/jsonSchema/nangoModelToJsonSchema.ts (hunk 1 above)
Comment (kaposke) [reply packages/utils/lib/jsonSchema/nangoModelToJsonSchema.ts:156]: As far as I tested there's nothing in nango.yaml that gets converted to `Record<any, any>`
